### ISSUE TITLE
Duration of the session is very short

### ISSUE THREAD

**Issue by eapl-gemugami** [sorenpeter/timeline#36]
I'm watching in the sandbox that the session lasts less than a day. I'm not sure of the exact duration, although every morning it asks me to login again.
https://twtxt-sandbox.eapl.me/timeline/

Based on this advice, I think 100 days (about 3 months) is a good compromise between usability and security:
https://auth0.com/blog/balance-user-experience-and-security-to-retain-customers/

**Comment by aelaraji** [comment]
I can't say I'm experiencing the same thing, I'm still logged into my instance since I had it installed, never been logged off unless I do so manually or restart the container serving it.

Could it be some browser preference deleting cookies/website data after a while? 
I'll try your instance if you don't mind and see if I get the same result.

**Comment by eapl-gemugami** [comment]
Thanks @aelaraji! As you say, it could be a privacy preference in my browser, although I've tried on Chrome and Zen, so let's see...

**Comment by aelaraji** [comment]
@eapl-gemugami just want to let you know, I've logged into your sandbox instance earlier and it has logged me out and it's been less than 8 hours since last time I checked. I'm still logged into mine, I don't think that has anything to do with the browser.

**Comment by eapl-gemugami** [comment]
Ok, so I think we could leave this issue open and work on something to make it longer.

In `twtxt-php` I had a 30 day session, I think that part wasn't integrated into `timeline`.
https://github.com/eapl-gemugami/twtxt-php/blob/master/session.php#L90

**Comment by sorenpeter** [comment]
Yeah, I've moved the original version of `session.php` to a `_wip_todo` folder, so that means it's not included in current version of timeline.
https://github.com/sorenpeter/timeline/blob/d13e72d11cbd5abb228bb2eac887e6895b943a02/_wip_todo/session.php#L60

I believe I did this because i could not understand how login via TOTP works, so make a simpler version of session.php.

It also frustrates me that the sessions are so short, so I'm all for reintroducing the 30 days session.

**Comment by eapl-gemugami** [comment]
Ok, I'm on vacations for the next few weeks, so I could help with it.

Also, although TOTP/Dynamic Passwords are nice ([Promoting the use of dynamic passwords](https://text.eapl.mx/promoting-the-use-of-dynamic-passwords)), I think nowadays Webauthn/Passkeys is more adequate and supported, if y'all see some value we could replace TOTP with it.

**Comment by eapl-gemugami** [comment]
I'm moving the about TOTP/Passkeys to discussions.

**Comment by eapl-gemugami** [comment]
In #44 I implemented a 'persistent session', based on my previous approach in twtxt-php, but using a different algorithm as in https://github.com/stevencorona/SessionHandlerCookie, since the previous one was incomplete.

I've tested every page, manually checking some expiration scenarios and seems to work OK in every case.
Could you also help me to check if it's working on your live sites?

For now session length is hard-coded to 30 days, but could be added to the `config.ini` if you see value on having a variable amount of days.
